### ISSUE TITLE
GPO "wish": Set own app.support.baseURL

### ISSUE THREAD

**Issue by HeiHei90** [mozilla/policy-templates#77]
Again, in environments where the clients are not directly connected to the internet...

It would be helpful, if you cold set your own help URL via GPO.

Thanks and kind Regards

**Comment by HeiHei90** [comment]
Maybe I am bad at searching stuff, but I didn't find anything about it. Is there a Tutorial where I can learn how to create a FireFox GPO and support you?

**Comment by mkaply** [comment]
> Maybe I am bad at searching stuff, but I didn't find anything about it. Is there a Tutorial where I can learn how to create a FireFox GPO and support you?

The way I learned the ADMX side was by looking at Google's ADMX files and reading the Microsoft documentation here:

https://docs.microsoft.com/en-us/windows/client-management/mdm/understanding-admx-backed-policies#a-href-idadmx-files-and-the-group-policy-editoraadmx-files-and-the-group-policy-editor

There are two components to this, implementing the policy on the Firefox side and then implementing on the ADMX side.

You can take a look at the patch in this bug:

https://bugzilla.mozilla.org/show_bug.cgi?id=1429179

to see implementing the policy on the Firefox side. We've made it very straightforward to add preference based policies including a test.

**Comment by HeiHei90** [comment]
For this Issue it should be smth. like this?

Policies.jsm

  "AppSupportBaseUrl": {
    let BaseURL = param.URL.spec;
    onBeforeUIStartup(manager, param) {
      if (param) {
        setAndLockPref("app.support.baseURL", BaseURL);
      }
    }
  },

policies-schema.json

    "AppSupportBaseURL": {
      "description": "Sets the app.support.baseURL",
      "first_available": "60.0",
      "enterprise_only": true,

      "type": "object",
      "properties": {
        "URL": {
          "type": "URL"
        }
      "required": ["URL"]
    },

firefox.adml

 <string id="AppSupportBaseURL">URL for Help Page</string>
      <string id="AppSupportBaseURL_Explain">If this policy is enabled, you can set your own help page. This configuration will be locked. Note: There will be automatically "firefox-help" added to your URL. 

If this policy is disabled or not configured, the standard help page will be used.</string>

firefox.admx

    <policy name="AppSupportBaseURL" class="Both" displayName="$(string.AppSupportBaseURL)" explainText="$(string.AppSupportBaseURL_Explain)" key="Software\Policies\Mozilla\Firefox" presentation="$(presentation.AppSupportBaseURL)" >
      <parentCategory ref="firefox" />
      <supportedOn ref="SUPPORTED_WINXPSP2" />
      <elements >
        <text id="AppSupportBaseURL" valueName="URL" required="true" />
       </elements>
    </policy>

**Comment by mkaply** [comment]
This actually is pretty good to what we would need to do.

Do you think we should update the base URL or add a new item to the help menu with a custom URL and name?

**Comment by HeiHei90** [comment]
The problem is: There are many links in this support/help-section, that would need to be changed.
Some of them seem not to be editable via about:config.
It would be great, if there were the possibility to add a new item to the help menu and disable all other entries, except "About" and "Starting with disabled AddOns" (Sorry, I am using the german version, it might be called differently).

**Comment by mkaply** [comment]
I agree. I'm thinking about a policy to enable/disable specific help items and add one help menu for your company.

**Comment by HeiHei90** [comment]
Name: Create custom help page menu entry
Explain: If this policy is enabled, you create a new menu entry at(? in?) category "Help". You can define the URL as you like. You can also lock all other Help Menu entries except "About Firefox" and "Restart with Add-ons disabled".

Something like this?

**Comment by DiecoTheBxxst** [comment]
There's a reson policy folder do not replicate between domain controller?

**Comment by HeiHei90** [comment]
@mkaply anything I can do to support you?

**Comment by HeiHei90** [comment]
If i understand this correctly, we have to create a userChrome.css to "hide" unnecessary menu entries?

**Comment by mkaply** [comment]
> If i understand this correctly, we have to create a userChrome.css to "hide" unnecessary menu entries?

No, we already have policies to disable all other entries on the dropdown.

I'll be doing new policy work in the next week or so.

**Comment by HeiHei90** [comment]
Great. :)
